src/posting/collection.py

```diff
@@ -246,12 +264,18 @@ def apply_template(self, variables: dict[str, Any]) -> None:
     def to_httpx(self, client: httpx.AsyncClient) -> httpx.Request:
         """Convert the request model to an httpx request."""
         headers = httpx.Headers(
-            [(header.name, header.value) for header in self.headers if header.enabled]
+            {
+                header.name: header.value
+                for header in self.headers
+                if header.enabled and header.name.lower() != "content-type"
+            }
```

Comment:
A list of tuples were being used here because you can have multiple headers with the same name and different values in a request. For example:

```http
Accept: text/html
Accept: application/xhtml+xml
Accept: application/xml;q=0.9
```